test/factories/accounts/user_factory.ex

```diff
@@ -10,14 +10,17 @@ defmodule Cadet.Accounts.UserFactory do
       def user_factory do
         %User{
           name: Faker.Name.En.name(),
-          role: Enum.random(Role.__enum_map__())
+          role: Enum.random(Role.__enum_map__()),
+          # Due to unique nusnet id constraint
+          nusnet_id: "E0" <> Integer.to_string(Enum.random(100_000..999_999))
```

Comment:
You should use `sequence` instead to ensure unique constraint. `Enum.random` might still result in a repetition. Idem for `student_factory`